src/Ogooreck.Sample.BusinessLogic.FSharp.Tests/Deciders/BankAccountTests.fs

```diff
@@ -0,0 +1,109 @@
+module BankAccountTests
+
+open System
+open Deciders.BankAccount
+open Ogooreck.BusinessLogic
+open Deciders.BankAccountDecider
+open Xunit
+
+
+let random = Random()
+let now = DateTimeOffset.UtcNow
+
+let getNow = fun () -> now
+
+let spec = Specification.For(decide getNow, evolve)
+
+let BankAccountOpenedWith =
+    fun bankAccountId now version ->
+        let accountNumber = Guid.NewGuid().ToString()
+
+        let clientId = Guid.NewGuid()
+        let currencyISOCode = "USD"
+
+        { BankAccountId = bankAccountId
+          AccountNumber = accountNumber
+          ClientId = clientId
+          CurrencyISOCode = currencyISOCode
+          CreatedAt = now
+          Version = version }
+        |> Event.BankAccountOpened
+
+let BankAccountClosedWith =
+    fun bankAccountId now version ->
+        { BankAccountId = bankAccountId
+          Reason = Guid.NewGuid().ToString()
+          ClosedAt = now
+          Version = version }
+        |> Event.BankAccountClosed
+
+[<Fact>]
+let ``GIVEN non existing bank account WHEN open with valid params THEN bank account is opened`` () =
+    let bankAccountId = Guid.NewGuid()
```

Comment:
FSCheck is super easy to use, and does all this for you.

Reply:
Sweet! Added in https://github.com/oskardudycz/Ogooreck/pull/13/commits/891d75f5f29e96ffa67c016cb56f25c858f3b3a6 😄